src/combinatorics/quiver/covering.lean

```diff
@@ -0,0 +1,257 @@
+end
+
+/--
+The star of the symmetrification of a quiver at a vertex `u` is equivalent to the sum of the star
+and the costar at `u` in the original quiver.
+ -/
+@[simps] def quiver.symmetrify_star (u : U) :
+  quiver.star (symmetrify.of.obj u) ≃ quiver.star u ⊕ quiver.costar u :=
+begin
+  fsplit,
+  { rintro ⟨v, (f|g)⟩, exacts [sum.inl ⟨v, f⟩, sum.inr ⟨v, g⟩], },
+  { rintro (⟨v, f⟩|⟨v, g⟩), exacts [⟨v, f.to_pos⟩, ⟨v, g.to_neg⟩], },
+  { rintro ⟨v, (f|g)⟩, simp, },
+  { rintro (⟨v, f⟩|⟨v, g⟩), simp, },
+end
+
+@[simp] lemma quiver.symmetrify_star_lapply {u v : U} (e : u ⟶ v) :
+  quiver.symmetrify_star u ⟨v, sum.inl e⟩ = sum.inl ⟨v, e⟩ := rfl
+
+@[simp] lemma quiver.symmetrify_star_rapply {u v : U} (e : v ⟶ u) :
+  quiver.symmetrify_star u ⟨v, sum.inr e⟩ = sum.inr ⟨v, e⟩ := rfl
+
+/--
+The costar of the symmetrification of a quiver at a vertex `u` is equivalent to the sum of the
+costar and the star at `u` in the original quiver.
+ -/
+@[simps] def symmetrify_costar (u : U) :
+  quiver.costar (symmetrify.of.obj u) ≃ quiver.costar u ⊕ quiver.star u :=
+begin
+  fsplit,
+  { rintro ⟨v, (f|g)⟩, exacts [sum.inl ⟨v, f⟩, sum.inr ⟨v, g⟩], },
+  { rintro (⟨v, f⟩|⟨v, g⟩), exacts [⟨v, quiver.hom.to_pos f⟩, ⟨v, quiver.hom.to_neg g⟩], },
+  { rintro ⟨v, (f|g)⟩, simp, },
+  { rintro (⟨v, f⟩|⟨v, g⟩), simp, },
+end
+
+lemma prefunctor.symmetrify_star (u : U) : φ.symmetrify.star u =
+ (quiver.symmetrify_star (φ.obj u)).symm ∘ sum.map (φ.star u) (φ.costar u) ∘
+ quiver.symmetrify_star u :=
+begin
+  rw equiv.eq_symm_comp,
+  ext ⟨v, (f|g)⟩;
+  simp,
+end
+
+protected lemma prefunctor.symmetrify_costar (u : U) : φ.symmetrify.costar u =
+ (symmetrify_costar (φ.obj u)).symm ∘ sum.map (φ.costar u) (φ.star u) ∘ symmetrify_costar u :=
+begin
+  rw equiv.eq_symm_comp,
+  ext ⟨v, (f|g)⟩;
+  simp,
+end
+
+protected lemma prefunctor.is_covering.symmetrify (hφ : φ.is_covering) : φ.symmetrify.is_covering :=
+begin
+  refine ⟨λ u, _, λ u, _⟩;
+  simp only [φ.symmetrify_star u, φ.symmetrify_costar u,
+             equiv_like.comp_bijective, equiv_like.bijective_comp],
+  exacts [⟨function.injective.sum_map (hφ.left u).left (hφ.right u).left,
+           function.surjective.sum_map (hφ.left u).right (hφ.right u).right⟩,
+          ⟨function.injective.sum_map (hφ.right u).left (hφ.left u).left,
+           function.surjective.sum_map (hφ.right u).right (hφ.left u).right⟩],
+end
+
+/-- The path star at a vertex `u` is the type of all paths starting at `u`. -/
+@[reducible] def quiver.path_star (u : U) := Σ v : U, path u v
+
+/-- A prefunctor induces a map of path stars. -/
+def prefunctor.path_star (u : U) : quiver.path_star u → quiver.path_star (φ.obj u) :=
+λ p, ⟨φ.obj p.1, φ.map_path p.2⟩
+
+@[simp] lemma prefunctor.path_star_apply {u v : U} (p : path u v) :
+  φ.path_star u ⟨v, p⟩ = ⟨φ.obj v, φ.map_path p⟩ := rfl
+
+theorem prefunctor.path_star_bijective (hφ : φ.is_covering) (u : U) :
+  function.bijective (φ.path_star u) :=
```

Comment:
Instead of assuming that there are inverses to the `φ.star u` and `φ.costar u` and showing that there's an inverse to the `φ.path_star u`, can you instead *construct* the inverses to the `φ.path_star u` from inverses for the `φ.star u` and `φ.costar u`?

Reply:
Probably can, but what's the gain?

Reply:
Yeah, I'm hoping for a cleaner proof. But I don't know how much it will make you change your framework.

Reply:
I'll try for a bit and see if helps and is reasonably easy, and give up otherwise.